src/main/java/org/osc/controller/nsfc/utils/RedirectionApiUtils.java

```diff
@@ -220,6 +228,42 @@ public InspectionHookEntity findInspHookByInspectedAndPort(NetworkElement inspec
         });
     }
 
+    public NetworkElementEntity retrieveNetworkElementFromOS(String portId, String portPairId) {
+        if (portId == null) {
+            return null;
+        }
+
+        Port port = this.osClient.networking().port().get(portId);
+
+        if (port == null) {
+            LOG.error("Port {} not found on openstack {}", portId, this.osClient.getEndpoint());
+            return null;
+        }
+
+        List<String> ips = emptyList();
+        if (port.getFixedIps() != null) {
+            ips = port.getFixedIps().stream().map(ip -> ip.getIpAddress()).collect(Collectors.toList());
+        }
+
+        return new NetworkElementEntity(port.getId(), ips, singletonList(port.getMacAddress()), portPairId);
+    }
+
+    public void validatePPGList(List<NetworkElement> portPairGroups) {
+        List<PortPairGroupEntity> ppgList = new ArrayList<>();
+
+        for (NetworkElement ne : portPairGroups) {
+            throwExceptionIfNullElementAndId(ne, "Port Pair Group Id");
+            PortPairGroupEntity ppg = findByPortPairgroupId(ne.getElementId());
```

Comment:
do we need to do this? would the sfc api handle it for us.

i undersatnd basic validation like throwExceptionIfNullElementAndId but would not expect making calls to openstack to validate